packages/component/src/SendBox/UploadButton.tsx

```diff
@@ -111,13 +123,21 @@ const UploadButton: FC<UploadButtonProps> = ({ className }) => {
 
   const handleFileChange = useCallback(
     ({ target: { files } }) => {
+      setFiles(files);
+
+      if (combineAttachmentsAndText) {
+        current.blur();
+        focus('sendBox');
+        return;
+      }
+
       sendFiles(files);
```

Comment:
Should use the `SendBoxComposer.useSubmit()` hook to send files. This will simplify lots of code.